### PR TITLE
applyAmpConfig: also update sourcemaps

### PR DESCRIPTION
**Pull request by samouri**
**summary**
`applyAmpConfig()` prepends `AMP_CONFIG` to AMP "primary" binaries like `v0.js`. Unfortunately we don't also update the sourcemaps, and instead have a hacky `preamble` for terser to update it for us.

In the ideal case, we would simply give the actual AMP_CONFIG to terser (or a different build tool) instead of just a `;`. To do that would require more refactor than I want to do right now.

I've opted to make `applyAmpConfig` also update the sourcemap file.

### CODE REVIEW

File: build-system/tasks/prepend-global/index.js [hunk 1]
```diff
@@ -108,7 +106,49 @@ function valueOrDefault(value, defaultValue) {
...
+  sanityCheck(fileString);
+  await writeTarget_(target, fileString, argv.dryrun);
+  const details =
+    '(' +
+    cyan(type) +
+    (opt_localDev ? ', ' + cyan('localDev') : '') +
+    (opt_fortesting ? ', ' + cyan('test') : '') +
+    (opt_derandomize ? ', ' + cyan('derandomized') : '') +
+    ')';
+  log('Applied AMP config', details, 'to', cyan(path.basename(target)));
```
Comment (rsimha) [inline build-system/tasks/prepend-global/index.js:138]: This message helps us ensure that we are applying the correct config to the correct files during `amp dist | build`. Now, the message isn't getting printed because we're using the banner / wrapper mechanism. I think it would be useful to move these two lines into `getConfig()`, and maybe reword the message _"Applied AMP config..."_ to _"Writing AMP config..."_. This way, both `applyConfig()` and `getAmpConfigForFile()` will print the same message under different use cases.

**Review by rsimha — COMMENTED** [review]
I like everything about this PR! I've suggested one change below to retain existing logging.

File: build-system/tasks/prepend-global/index.js (hunk 1 above)
Comment (samouri) [reply build-system/tasks/prepend-global/index.js:138]: Thank you for noticing this. I'll follow your advice here move it to `getConfig`. WDYT about the word _Creating AMP Config_ instead?

Comment (rsimha) [reply build-system/tasks/prepend-global/index.js]: Here's what the logging looks like today.

image

_"Creating AMP config (prod, localDev, test) to v0.js"_ sounds weird.

If you don't like _"Writing AMP config..."_, how about making it _"Applying AMP config..."_?

File: build-system/tasks/prepend-global/index.js [hunk 2]
```diff
@@ -155,20 +189,47 @@ async function applyConfig(
...
   const details =
     '(' +
-    cyan(config) +
+    cyan(type) +
     (opt_localDev ? ', ' + cyan('localDev') : '') +
     (opt_fortesting ? ', ' + cyan('test') : '') +
     (opt_derandomize ? ', ' + cyan('derandomized') : '') +
     ')';
-  log('Applied AMP config', details, 'to', cyan(path.basename(target)));
+  log('Created AMP config', details, 'to', cyan(path.basename(target)));
```
Comment (rsimha) [inline build-system/tasks/prepend-global/index.js:200]: Changed `Created` to `Applying` and removed `path.basename()` so the extension gets printed.

```suggestion
  log('Applying AMP config', details, 'to', cyan(target));
```
